cli-paho-java/src/main/java/com/redhat/mqe/amc/Sender.java

```diff
@@ -68,7 +68,24 @@ public void startClient() throws MqttException {
         try {
             sender = new MqttClient(cliBroker, cliClientId, persistence);
             log.fine("Connecting to broker: " + broker);
-            sender.connect(setConnectionOptions(new MqttConnectOptions()));
+
+            MqttConnectOptions connectOptions = new MqttConnectOptions();
+
+            if (cliWillFlag == 1) {
+                if (cliWillDestination.isEmpty()) {
+                    log.severe("Will destination cannot be empty.");
+                    System.exit(0);
+                }
+
+                if (cliWillMessage.isEmpty()) {
+                    log.severe("Will message cannot be empty.");
+                    System.exit(0);
+                }
+
+                connectOptions.setWill(cliWillDestination, cliWillMessage.getBytes(), cliWillQos, cliWillRetained);
```

Comment:
The will message relates to all three types of client (sender, receiver, (connector). Not only sender. Please add this to the receiver as well. Receivers can set last will message too.